src/Evolution/Systems/Cce/AnalyticSolutions/RobinsonTrautman.cpp

```diff
@@ -0,0 +1,404 @@
+  for (size_t i = 0; i < std::min(initial_modes_.size(), goldberg_modes.size());
+       ++i) {
+    goldberg_modes.data()[i] = gsl::at(initial_modes_, i);
+  }
+
+  SpinWeighted<ComplexModalVector, 0> libsharp_modes{
+      Spectral::Swsh::size_of_libsharp_coefficient_vector(l_max_)};
+  const auto& coefficients_metadata =
+      Spectral::Swsh::cached_coefficients_metadata(l_max_);
+  for (const auto& coefficient : coefficients_metadata) {
+    Spectral::Swsh::goldberg_modes_to_libsharp_modes_single_pair(
+        coefficient, make_not_null(&libsharp_modes), 0,
+        goldberg_modes.data()[Spectral::Swsh::goldberg_mode_index(
+            l_max_, coefficient.l, static_cast<int>(coefficient.m), 0)],
+        goldberg_modes.data()[Spectral::Swsh::goldberg_mode_index(
+            l_max_, coefficient.l, -static_cast<int>(coefficient.m), 0)]);
+  }
+
+  // re-use the `dense_output_rt_scalar_` buffers to reduce allocations
+  Spectral::Swsh::inverse_swsh_transform(
+      l_max_, 1, make_not_null(&get(dense_output_rt_scalar_)), libsharp_modes);
+  // offset the scalar to 'reasonable' value
+  get(dense_output_rt_scalar_).data() =
+      1.0 + std::complex<double>(1.0, 0.0) *
+                real(get(dense_output_rt_scalar_).data());
+  du_rt_scalar(make_not_null(&get(dense_output_du_rt_scalar_)),
+               get(dense_output_rt_scalar_));
+  stepper_ = boost::numeric::odeint::make_dense_output(
+      tolerance_ * 0.01, tolerance_,
+      boost::numeric::odeint::runge_kutta_dopri5<ComplexDataVector>{});
+  const double rt_scalar_scale = max(abs(get(dense_output_rt_scalar_).data()));
+  const double du_rt_scalar_scale =
+      max(abs(get(dense_output_du_rt_scalar_).data()));
+
+  double initial_step = extraction_radius_ * 1.0e-4 / square(l_max_);
+  if (rt_scalar_scale > 1.0e-5 and du_rt_scalar_scale > 1.0e-5) {
+    initial_step =
+        0.1 * rt_scalar_scale / (du_rt_scalar_scale * square(l_max_));
+  }
+  stepper_.initialize(get(dense_output_rt_scalar_).data(), start_time_,
+                      initial_step);
+
+  const auto rt_system = [this](const ComplexDataVector& local_rt_scalar,
+                                ComplexDataVector& local_du_rt_scalar,
+                                const double /*t*/) noexcept {
+    local_du_rt_scalar.destructive_resize(local_rt_scalar.size());
+    const SpinWeighted<ComplexDataVector, 0> rt_scalar_reference;
+    make_const_view(make_not_null(&rt_scalar_reference.data()), local_rt_scalar,
+                    0, local_rt_scalar.size());
+    SpinWeighted<ComplexDataVector, 0> du_rt_scalar_reference;
+    du_rt_scalar_reference.set_data_ref(local_du_rt_scalar.data(),
+                                        local_du_rt_scalar.size());
+    du_rt_scalar(make_not_null(&du_rt_scalar_reference), rt_scalar_reference);
+    Spectral::Swsh::filter_swsh_boundary_quantity(
+        make_not_null(&du_rt_scalar_reference), l_max_, l_max_ - 3);
```

Comment:
Should the `3` be an input parameter? some docs or comments on where it comes from would be helpful.

Reply:
I'll add documentation; I don't think this filtering parameter will be productive to make an input argument for a test solution. No filtering is dangerous, but it is rare that more filtering is desired than the top two modes.